.github/workflows/rust.yaml

```diff
@@ -0,0 +1,25 @@
+name: Rust
+
+on:
+  push:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Build
+      run: cargo build --verbose
+    - name: Run tests
+      run: cargo test --verbose
+    - name: Run Clippy
+      run: cargo clippy
```

Comment:
For clippy to take the warnings as errors, as it should be in the CI, you need to add `-- -D warnings`